haskell/test/Lorentz/Contracts/Test/Management.hs

```diff
@@ -0,0 +1,504 @@
+        withSender admin $ lCallEP stablecoinContract (Call @"Change_master_minter") wallet2
+        validate . Right $ expectAnySuccess
+
+    it "transferring ownership fails if sender is not contract owner" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender wallet1 $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet1
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "fails if previous contract owner tries to use ownership privileges" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet1
+        withSender wallet1 $ lCallEP stablecoinContract (Call @"Accept_ownership") ()
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet2
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "accepting ownership fails if sender is not pending contract owner" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet1
+        withSender wallet2 $ lCallEP stablecoinContract (Call @"Accept_ownership") ()
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+    it "transfer ownership can be called multiple times each of which invalidates the previous call" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet1
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet2
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet3
+        withSender wallet3 $ lCallEP stablecoinContract (Call @"Accept_ownership") ()
+        validate . Right $ expectAnySuccess
+
+    it "contract cannot retain ownership privileges if pending owner was changed by subsequent transfer ownership call" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet1
+        withSender admin $ lCallEP stablecoinContract (Call @"Transfer_ownership") wallet2
+        withSender wallet1 $ lCallEP stablecoinContract (Call @"Accept_ownership") ()
+        validate . Left $ lExpectAnyMichelsonFailed stablecoinContract
+
+
+
+  describe "Master minter" $ do
+    it "changes contract master minter properly" $ integrationalTestExpectation $ do
+      withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
```

Comment:
I think origination function should accept admin, pauser and master minter addresses for these tests to really make sense.

Reply:
The point I was trying to make is to simplify things as possible by making only one superuser. I've changed its name if this clarifies some things. Also, I've added the capability of changing roles during origination, but it still seems to be unused.

Reply:
I am not sure if we can test that the roles work well if you assign all roles to a single address.

>but it still seems to be unused.

Didn't get you.

Reply:
I mean that previously we could not change originated params and had set up default permissions during origination. While testing we still change roles by calling specific entrypoints and validating it after by using specific patterns. It's not essential tho, just mentioning.